pkg/secretless/entrypoint/start_secretless.go

```diff
@@ -0,0 +1,124 @@
+package entrypoint
```

Comment:
I don't think I've seen naming like this in other codebases nor in ours (`start_secretless` => `<verb>_<type>`) so I'm wondering what might be the reasoning behind such a large deviation from the norm? On face value of what I'm reading, the barrier for using a `.go` file named like that along with a few other public methods that do not seem to correlate too closely seems to be not enough for a deviation but I'd like to hear the other side of the argument on that.

Reply:
Ok so this one is a bit of a longer explanation.  First off, I should say that thing I'm currently calling `Secretless` should be renamed to `StartProxyServices` (for reasons I'll explain shortly).

This is based on a naming convention which I find to be one of the most helpful naming conventions I've ever encountered.  At a high-level, the convention is just:

- Name functions that return things after the noun they return.  Use "AdjectiveNoun" when needed for clarity.
- Name functions that _do_ things using a verb phrase describing what they _do_.  "Do" here means "do something with side effects".  Eg, "CreateUser" (adds user to a db) "StartServiceX" (starts listening on a network), "SaveConfigFile" (writes a file to disk), etc.

So where does this leave us with `StartSecretless`.  Well `StartSecretless` is a "function object".  That is, it's a type that takes in some side-effectful dependencies (injected via ctor), and is the "run" or "executed" (sometimes with additional runtime inputs, but not in this case).  The method that executes the function object is typically called `Run()` or `Call()` or in this case `Start()` (`Start()` is probably a mistake, actually).  Thus the naming convention for a function object is the same as for a side-effectful function.  

More on this technique (in the context of ruby, though that's not really relevant) [here](https://www.icelab.com.au/notes/functional-command-objects-in-ruby/).  I've used these conventions for the `authn` stuff in Conjur and have been happy with the results.

What might be even cleaner than how I have it now is for `StartSecretless` to return a type that has nothing but a `Call()` method, and for the `Call()` method to return a `Stopper` function.  Anyway, what do you think -- both about all this and the specific question I just asked?